ViSearchSDK/ViSearchSDK/Classes/ViProductSearch.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  ViProductSearch.swift
+//  ViSearchSDK
+//
+//  Created by visenze on 5/3/21.
+//
+
+import Foundation
+import ViSenzeAnalytics
+
+/// Singleton/Shared instance wrapper for various APIs
+open class ViProductSearch : NSObject {
+    public static let BASE_URL = "https://search.visenze.com"
+    public static let SBI_ENDPOINT = "/v1/product/search_by_image"
+    public static let VSR_ENDPOINT = "/v1/product/search_by_id"
+    
+    public static let sharedInstance = ViProductSearch();
+    
+    /// App Key for search-only access
+    private var appKey : String? = nil
+    
+    /// Placement ID
+    private var placementId : Int? = nil
+    
+    private var client : ViProductSearchClient? = nil
+    
+    /// Default Constructor
+    private override init() {
+        super.init()
+    }
+    
+    /// Set up the SDK with the proper authentications, needs to be called first prior to any other SDK
+    /// functions
+    ///
+    /// - parameter appKey: App key
+    /// - parameter placementId: Placement ID
+    public func setUp(appKey:String, placementId:Int) -> Void {
```

Comment:
is this a static function? the constructor is private, how are the users calling this?